lua/lspconfig/configs.lua

```diff
@@ -40,10 +40,11 @@ function configs.__newindex(t, config_name, config_def)
       commands = { config.commands, 't', true },
     }
     if config.commands then
-      for k, v in pairs(config.commands) do
+      vim.deprecate('lspconfig.[server].setup.commands', 'vim.api.nvim_create_user_command', '0.9', 'nvim-lspconfig')
```

Comment:
- soft deprecation is correct. Only show a warning/error, but continue to function
- as others mentioned, don't change the structure. we don't want a breaking change yet.